hibernate-core/src/main/java/org/hibernate/sql/ast/tree/select/SortSpecification.java

```diff
@@ -20,18 +20,15 @@
 	private final Expression sortExpression;
 	private final SortOrder sortOrder;
 	private final NullPrecedence nullPrecedence;
+	private final Collation collation;
 
-	public SortSpecification(Expression sortExpression, String collation, SortOrder sortOrder) {
-		this( sortExpression, collation, sortOrder, NullPrecedence.NONE );
+	public SortSpecification(Expression sortExpression, SortOrder sortOrder) {
+		this( sortExpression, null, sortOrder, NullPrecedence.NONE );
 	}
 
 	public SortSpecification(Expression sortExpression, String collation, SortOrder sortOrder, NullPrecedence nullPrecedence) {
-		if ( collation == null ) {
-			this.sortExpression = sortExpression;
-		}
-		else {
-			this.sortExpression = new Collate( sortExpression, collation );
-		}
```

Comment:
@beikov 

FTR, this is what you were missing: this was what it did before.

The client code which instantiates this class all passes in a collation. I don't know why, and I don't know what that was for. Quite likely, it can all be ripped out now, but I certainly was not going to rip out code I don't understand.